### ISSUE TITLE
Support custom generated source extensions

### ISSUE THREAD

**Issue by kirbyfan64** [dart-lang/sdk#33750]
Since URI_HAS_NOT_BEEN_GENERATED has been added 2 years ago, there is still [a TODO to generalize the mechanism for finding generated sources](https://github.com/dart-lang/sdk/blame/master/pkg/analyzer/lib/src/task/dart.dart#L5675). Now that everything is being migrated over to the build package, where everything has to have a new extensions, it would be nice to have this generalized somehow, even if it's just a `generated-source-extensions` option in `analysis_options.yaml`.

**Comment by bwilkerson** [comment]
@natebosch Is this still an issue for package:build generated files?

**Comment by natebosch** [comment]
yes, it's still possible to have files with any extension that can be produced during the build and show up before a build as an error.
